src/routes/SearchPage/index.js

```diff
@@ -66,23 +83,14 @@ class SearchPage extends React.Component {
   render () {
     return (
       <RichLayout location={this.props.location}>
-        <PageFetcher>
-          <Search className={style.searchSpacing}
-                  filterText={this.state.filterText}
-                  onFilterTextChange={(filterText) => this.setState({filterText: (filterText)})}
-          />
-          <ContentListAdapter location={this.props.location} language={this.props.language} filterText={this.state.filterText}/>
-        </PageFetcher>
+        <Search className={style.searchSpacing}
+                filterText={this.state.filterText}
+                onFilterTextChange={(filterText) => this.setState({filterText: (filterText)})}
+        />
+        <FetchingContentWrapper filterText={this.state.filterText} location={this.props.location}/>
       </RichLayout>
     )
   }
 }
 
-function mapStateToProps (state) {
-  return {
-    location: state.router.params.location,
-    language: state.router.params.language
-  }
-}
-
-export default connect(mapStateToProps)(SearchPage)
+export default connect((state) => ({location: state.router.params.location}))(SearchPage)
```

Comment:
We don't need location in the SearchPage props, if you add the location-prop in the mapStateToProps for the FetchingContentWrapper. Also the RichLayout is connected to the state itself.
So we disconnect the SearchPage from the state 🎉

Reply:
done